pkg/controller/cognitoidentityprovider/identityprovider/setup.go

```diff
@@ -0,0 +1,163 @@
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type custom struct {
+	kube     client.Client
+	client   svcsdkapi.CognitoIdentityProviderAPI
+	external *external
+	resolver cognitoidentityprovider.ResolverService
+}
+
+func preObserve(_ context.Context, cr *svcapitypes.IdentityProvider, obj *svcsdk.DescribeIdentityProviderInput) error {
+	obj.ProviderName = awsclients.String(meta.GetExternalName(cr))
+	obj.UserPoolId = cr.Spec.ForProvider.UserPoolID
+	return nil
+}
+
+func preDelete(_ context.Context, cr *svcapitypes.IdentityProvider, obj *svcsdk.DeleteIdentityProviderInput) (bool, error) {
+	obj.ProviderName = awsclients.String(meta.GetExternalName(cr))
+	obj.UserPoolId = cr.Spec.ForProvider.UserPoolID
+	return false, nil
+}
+
+func postObserve(_ context.Context, cr *svcapitypes.IdentityProvider, obj *svcsdk.DescribeIdentityProviderOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	cr.SetConditions(xpv1.Available())
+
+	return obs, nil
+}
+
+func (e *custom) preCreate(ctx context.Context, cr *svcapitypes.IdentityProvider, obj *svcsdk.CreateIdentityProviderInput) error {
+	obj.UserPoolId = cr.Spec.ForProvider.UserPoolID
+	obj.ProviderName = awsclients.String(meta.GetExternalName(cr))
+
+	if cr.Spec.ForProvider.ProviderDetails != nil {
+		obj.ProviderDetails = cr.Spec.ForProvider.ProviderDetails
+	} else {
+		providerDetails, err := e.resolver.GetProviderDetails(ctx, e.kube, &cr.Spec.ForProvider.ProviderDetailsSecretRef)
+		if err != nil {
+			return err
+		}
+		obj.ProviderDetails = providerDetails
+	}
+	return nil
+}
+
+func (e *custom) preUpdate(ctx context.Context, cr *svcapitypes.IdentityProvider, obj *svcsdk.UpdateIdentityProviderInput) error {
+	obj.UserPoolId = cr.Spec.ForProvider.UserPoolID
+	obj.ProviderName = awsclients.String(meta.GetExternalName(cr))
+	if cr.Spec.ForProvider.ProviderDetails != nil {
+		obj.ProviderDetails = cr.Spec.ForProvider.ProviderDetails
+	} else {
+		providerDetails, err := e.resolver.GetProviderDetails(ctx, e.kube, &cr.Spec.ForProvider.ProviderDetailsSecretRef)
+		if err != nil {
+			return err
+		}
+		obj.ProviderDetails = providerDetails
+	}
+	return nil
+}
+
+func (e *custom) isUpToDate(cr *svcapitypes.IdentityProvider, resp *svcsdk.DescribeIdentityProviderOutput) (bool, error) {
+	provider := resp.IdentityProvider
+	providerDetails := make(map[string]*string)
+	if cr.Spec.ForProvider.ProviderDetails == nil {
+		ctx := context.Background()
+		p, err := e.resolver.GetProviderDetails(ctx, e.kube, &cr.Spec.ForProvider.ProviderDetailsSecretRef)
+		if err != nil {
+			return false, err
+		}
+		providerDetails = p
+	}
+	switch {
+	case !reflect.DeepEqual(cr.Spec.ForProvider.AttributeMapping, provider.AttributeMapping),
+		!reflect.DeepEqual(cr.Spec.ForProvider.IDpIdentifiers, provider.IdpIdentifiers),
```

Comment:
`cr.Spec.ForProvider.IDpIdentifiers ` think this is related to ACK - that we have here `IDp` insted of `Idp` ?

Reply:
Yes, this is due to ACK